src/higley_lab_to_nwb/benisty_2024/benisty_2024_nwbconverter.py

```diff
@@ -1,12 +1,29 @@
 """Primary NWBConverter class for this dataset."""
 
-
+from typing import Dict
 from neuroconv import NWBConverter
-from neuroconv.datainterfaces import ScanImageMultiFileImagingInterface
+from neuroconv.datainterfaces import ScanImageMultiFileImagingInterface, Suite2pSegmentationInterface
+from neuroconv.utils import DeepDict
 
 class Benisty2024NWBConverter(NWBConverter):
     """Primary conversion class."""
 
     data_interface_classes = dict(
         TwoPhotonImagingGreenChannel=ScanImageMultiFileImagingInterface,
-    )
\ No newline at end of file
+        SegmentationGreenChannel=Suite2pSegmentationInterface,
+    )
+    def __init__(self, source_data: Dict[str, dict],ophys_metadata: Dict[str, dict],  verbose: bool = True):
+        super().__init__(source_data, verbose)
+        self.ophys_metadata=ophys_metadata
+
+    def get_metadata(self) -> DeepDict:
+        metadata = super().get_metadata()
+        segmentation_metadata = self.data_interface_objects["SegmentationGreenChannel"].get_metadata()
+        for segmentation_metadata_ind in range(len(segmentation_metadata["Ophys"]["ImageSegmentation"]["plane_segmentations"])):
+            metadata["Ophys"]["ImageSegmentation"]["plane_segmentations"][segmentation_metadata_ind]["imaging_plane"] = self.ophys_metadata["Ophys"]["ImagingPlane"][0]["name"]
+        
+        metadata["Ophys"]["Device"] = self.ophys_metadata["Ophys"]["Device"]
+        metadata["Ophys"]["TwoPhotonSeries"] = self.ophys_metadata["Ophys"]["TwoPhotonSeries"]
+        metadata["Ophys"]["ImagingPlane"] = self.ophys_metadata["Ophys"]["ImagingPlane"]
+
+        return metadata
```

Comment:
I think you can just use the imaging interface's metadata to override the segmentation interface's metadata, I don't see why you need to pass it at init.

Reply:
Because I get an error from `validate_metadata() `function that the `ophys_metadata` argument is missing. Should I rewrite also  `validate_metadata()` and `get_metadata_schema()` for the converter to propagate `ophys_metadata`

Reply:
> Because I get an error from `validate_metadata() `function that the `ophys_metadata` argument is missing. Should I rewrite also `validate_metadata()` and `get_metadata_schema()` for the converter to propagate `ophys_metadata`

Yes my question was why you need `ophys_metadata` in the first place. Originally what I meant in the comment above is to do something like this: https://github.com/catalystneuro/dombeck-lab-to-nwb/blob/ebf959970fcdc7996c2e06d97dadd4d0f9e2adca/src/dombeck_lab_to_nwb/nagappan_embargo_2023/nagappan_embargo_2023nwbconverter.py#L115

If you look at https://github.com/catalystneuro/dombeck-lab-to-nwb/blob/ebf959970fcdc7996c2e06d97dadd4d0f9e2adca/src/dombeck_lab_to_nwb/nagappan_embargo_2023/nagappan_embargo_2023nwbconverter.py#L107 this function, I'm using the metadata from the imaging interface to override the optical channel in the segmentation metadata. 

But this is fine for me if this works better for you.